shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -439,58 +427,85 @@ static BOOL IsApproximatelyEqual(float x, float y, float delta) {
 // Checks whether point should be considered closer to selectionRect compared to
 // otherSelectionRect.
 //
-// If checkRightBoundary is set, the right-center point on selectionRect and
-// otherSelectionRect will be used instead of the left-center point.
+// Uses the leading-center point on selectionRect and otherSelectionRect to compare.
+// For left-to-right text, this means the left-center point, and for right-to-left text,
+// this means the right-center point.
+//
+// If useTrailingBoundaryOfSelectionRect is set, the trailing-center point on selectionRect
+// will be used instead of the leading-center point.
 //
 // This uses special (empirically determined using a 1st gen iPad pro, 9.7" model running
 // iOS 14.7.1) logic for determining the closer rect, rather than a simple distance calculation.
 // First, the closer vertical distance is determined. Within the closest y distance, if the point is
 // above the bottom of the closest rect, the x distance will be minimized; however, if the point is
 // below the bottom of the rect, the x value will be maximized.
-static BOOL IsSelectionRectCloserToPoint(CGPoint point,
-                                         CGRect selectionRect,
-                                         CGRect otherSelectionRect,
-                                         BOOL checkRightBoundary) {
-  CGPoint pointForSelectionRect =
-      CGPointMake(selectionRect.origin.x + (checkRightBoundary ? selectionRect.size.width : 0),
-                  selectionRect.origin.y + selectionRect.size.height * 0.5);
+static BOOL IsSelectionRectBoundaryCloserToPoint(CGPoint point,
+                                                 CGRect selectionRect,
+                                                 BOOL selectionRectIsRTL,
+                                                 BOOL useTrailingBoundaryOfSelectionRect,
+                                                 CGRect otherSelectionRect,
+                                                 BOOL otherSelectionRectIsRTL,
```

Comment:
don't we also need a `useTrailingBoundaryOfOtherSelectionRect`?

Reply:
The otherRect is always using leading boundary